instrumentation/kafka-streams/README.md

```diff
@@ -95,6 +95,25 @@ referencing the parent context stored on Headers, if available.
 Be aware that operations that require `builder.transformer(...)` will cause re-partitioning when
 grouping or joining downstream ([Kafka docs](https://kafka.apache.org/documentation/streams/developer-guide/dsl-api.html#applying-processors-and-transformers-processor-api-integration)).
 
+### Why not to trace _every_ Kafka Streams operation?
+
+When starting to design this instrumentation, “trace everything” was the first idea:
+When a message enters the Kafka Streams topology starts a new `poll` span, and every operation
+(e.g. `map`, `filter`, `join`, etc.) is chained as an additional child span.
+
+Kafka Streams materializes its topology _internally_ based on DSL operations. 
+Therefore, is not possible to hook into
+the topology creation process to instrument each operation.
+
+Even though this could be desirable it would require, first, to add new "doors" on the Kafka Streams
```

Comment:
move this sentence to the former paragraph to finish the thought? maybe as 

"We considered changing Kafka Streams to have hooks to do this, but it would be a hard sell. The maintainers of Kafka Streams would have to agree to add hooks in dozens of places, and move lacking context so that they can work"

^^ is an assumed rationale, you can correct it.